torch/onnx/symbolic_opset11.py

```diff
@@ -830,3 +830,100 @@ def prim_ConstantChunk(g, self, chunks, dim):
+                           'repeats rank.')
+    if repeats_sizes is None:
+        raise RuntimeError('Unsupported: ONNX export of repeat_interleave for unknown '
+                           'repeats size.')
+    if input_sizes is None:
+        raise RuntimeError('Unsupported: ONNX export of repeat_interleave for unknown '
+                           'input size.')
+
+    input_sizes_temp = input_sizes.copy()
+    perm_i = [0]
+    for idx, input_size in enumerate(input_sizes):
+        perm_i.append(idx + 1)
+        if input_size is None:
+            input_sizes[idx], input_sizes_temp[idx] = 0, -1
+    perm_i[0], perm_i[dim] = perm_i[dim], perm_i[0]
+
+    # Cases when repeats is a single value tensor and dim has unknown input size
+    if (repeats_dim == 0 or (repeats_dim == 1 and repeats_sizes[0] == 1)) and input_sizes[dim] == 0:
+        if not sym_help._is_tensor(repeats):
+            repeats = g.op("Constant", value_t=torch.LongTensor(repeats))
+        reps = sym_help._size_helper(g, input, dim)
+        reps = unsqueeze(g, reps, 0)
+        repeats = g.op("Expand", repeats, reps)
+    elif repeats_dim == 1 and input_sizes[dim] == 0:
+        reps = sym_help._size_helper(g, input, dim)
+        reps = unsqueeze(g, reps, 0)
+    else:
+        return torch.onnx.symbolic_opset9.repeat_interleave(g, self, repeats, final_dim)
+
+    reps_like = g.op("ConstantOfShape", g.op("Shape", repeats),
+                     value_t=torch.tensor([1], dtype=torch.long))
+    r_splits = split(g, repeats, reps_like, 0)
+    i_splits = split(g, input, reps_like, dim)
+
+    input_sizes[dim], input_sizes_temp[dim] = -1, 1
+
+    # Create a loop to iterate over each value along the dimension
+    # and perform individual interleaving using the repeats tensor
+    # Loop is of the following pattern
+    # input (trip_count, cond)
+    #   int trip_count = ...;
+    #   bool cond = ...;
+    #   for (int i=0; i < trip_count && cond; ++i) {
+    #     cond = ...;
+    #   }
+
+    # Loop conditions
+    loop_condition = g.op("Constant", value_t=torch.tensor(1))
+    loop_condition = g.op("Cast", loop_condition, to_i=9)
+    loop_len = reps
+    loop = g.op("Loop", loop_len, loop_condition)
+
+    # Loop inputs
+    loop_block = _add_block(loop.node())
+    block_input_iter = _add_input_to_block(loop_block)
+    cond = _add_input_to_block(loop_block)
+
+    r_split = loop_block.op("SequenceAt", r_splits, block_input_iter)
+    i_split = loop_block.op("SequenceAt", i_splits, block_input_iter)
+
+    i_split = unsqueeze(loop_block, i_split, dim + 1)
+    r_concat = [loop_block.op("Constant", value_t=torch.LongTensor(input_sizes_temp[:dim + 1])),
+                r_split,
+                loop_block.op("Constant", value_t=torch.LongTensor(input_sizes_temp[dim + 1:]))]
+    r_concat = loop_block.op("Concat", *r_concat, axis_i=0)
+    i_split = expand(loop_block, i_split, r_concat, None)
```

Comment:
can `expand` take `r_concat` that contain -1? what about multiple -1?

Reply:
Yes, it should be able to handle this, as values in input_sizes_temp are being set to -1